PathPlanning/DStarLite/DStarLite.py

```diff
@@ -0,0 +1,278 @@
+                    plt.plot(self.detected_obstacles_for_plotting_x,
+                             self.detected_obstacles_for_plotting_y, ".k")
+                    plt.pause(pause_time)
+            self.spoofed_obstacles.pop(0)
+
+        # Allows random generation of obstacles
+        random.seed()
+        if random.random() > 1 - p_create_random_obstacle:
+            x = random.randint(0, self.x_max - 1)
+            y = random.randint(0, self.y_max - 1)
+            new_obs = Node(x, y)
+            if compare_coordinates(new_obs, self.start) or \
+               compare_coordinates(new_obs, self.goal):
+                return changed_vertices
+            changed_vertices.append(Node(x, y))
+            self.detected_obstacles.append(Node(x, y))
+            if show_animation:
+                self.detected_obstacles_for_plotting_x.append(x)
+                self.detected_obstacles_for_plotting_y.append(y)
+                plt.plot(self.detected_obstacles_for_plotting_x,
+                         self.detected_obstacles_for_plotting_y, ".k")
+                plt.pause(pause_time)
+        return changed_vertices
+
+    def main(self, start: Node, goal: Node):
+        pathx = []
+        pathy = []
+        last = start
+        self.initialize(start, goal)
+        self.compute_shortest_path()
+        pathx.append(start.x)
+        pathy.append(start.y)
+        while not compare_coordinates(self.goal, self.start):
+            if self.g[start.x][start.y] == math.inf:
+                print("No path possible")
+                return False, pathx, pathy
+            self.start = min(self.succ(self.start),
+                             key=lambda sprime:
+                             self.c(self.start, sprime) +
+                             self.g[sprime.x][sprime.y])
+            pathx.append(self.start.x)
+            pathy.append(self.start.y)
+            if show_animation:
+                plt.plot(pathx, pathy, "-r")
+                plt.axis("equal")
+                plt.pause(pause_time)
+            changed_vertices = self.detect_changes()
+            if changed_vertices is not None:
+                self.km += self.h(last)
+                last = self.start
+                for u in changed_vertices:
+                    if compare_coordinates(u, self.start):
+                        continue
+                    self.update_vertex(u)
+                self.compute_shortest_path()
+        return True, pathx, pathy
+
+
+def main():
+    start = [0, 0]
+    goal = [5, 5]
+
+    obsx = [1, 3]
+    obsy = [1, 2]
+    obstacles = [Node(x, y) for x, y in zip(obsx, obsy)]
+
+    # Obstacles discovered at time = row
+    # time = 1, obstacles discovered at (0, 2), (9, 2), (4, 0)
+    # time = 2, obstacles discovered at (0, 1), (7, 6)
+    # ...
+    spoofed_obstacles = list()
+    spoofed_obstacles_x = [[0, 9, 4], [0, 7], [], [], [], [], [], [5]]
+    spoofed_obstacles_y = [[2, 2, 0], [1, 6], [], [], [], [], [], [4]]
```

Comment:
Could you use the same configuration of astar.py? 
https://github.com/AtsushiSakai/PythonRobotics/blob/1de0ffddb1ea11396a019403bbb986c8df8e0bdd/PathPlanning/AStar/a_star.py#L236-L263

Reply:
I can modify the start and end coordinates.

Some clarifications/context:
- **Negative grid coordinates**: The grid currently only takes indices >= 0 and makes a 1 to 1 correlation between grid coordinate and index for easy understanding.
- **Robot radius**: Adding robot radius will make it a little more complex to check for obstacles, and attempting simplicity
- **Boundary wall**: The boundary wall is not needed as the planner will not attempt to expand points not in the grid (though I can add in a check to ensure start and end points are on the grid)
- **Obstacles**: The obstacles were chosen specifically for forcing a reroute during the dynamic planning, and demonstrate how an example of discovering obstacles with time

Please let me know which of the changes referred to above (in bold) you would like me to make; will make them.